src/hotspot/share/opto/loopopts.cpp

```diff
@@ -367,6 +352,19 @@ void PhaseIdealLoop::dominated_by(IfProjNode* prevdom, IfNode* iff, bool flip, b
     if (cd->depends_only_on_test() && _igvn.no_dependent_zero_check(cd)) {
       assert(cd->in(0) == dp, "");
       _igvn.replace_input_of(cd, 0, prevdom);
+      if (range_check_predicate) {
+        // Loads and range check Cast nodes that are control dependent on this range check (that is about to be removed)
+        // now depend on multiple dominating range checks. After the removal of this range check, these control
+        // dependent nodes end up at the lowest/nearest dominating check in the graph. To ensure that these Loads/Casts
+        // do not float above any of the dominating checks (even when the lowest dominating check is later replaced by
+        // yet another dominating check), we need to pin them at the lowest dominating check.
```

Comment:
Should we also add this updated comment to `ifnode.cpp:569` and `ifnode.cpp:1536`?

Reply:
Yes, right. Thanks! I updated the change.

Reply:
Great, thanks!